### DISCUSSION TITLE
Chairs need picked required id!

### DISCUSSION THREAD

**Discussion by ApolloDraxium** [FakeFishGames/Barotrauma#11979] [Ideas and suggestions]
Chairs need an option in the barotrauma sub editor to be able to ID a chair to prevent other NPCs from using the chair.

So during a roleplay server, sometimes we end up getting crewmates or NPCs that are not supposed to be sitting in said chair.
Chairs used for high commanding officers or chairs used in meetings with designated spots are either taken by an NPC or a person who is there to cause fail RP. I know this is a bizarre suggestion but adding an ID restriction on the chairs would be a great benefit to the roleplaying society of Barotrauma. Thanks for reading my suggestion!

**Comment by Arctodites** [comment]
A lot of items could benefit from having an 'item required to interact' (i.e. picked required) field in the 'CONTROLLER' panel.  It would be especially useful for switches, levers, circuit boxes, terminals, and cabinets, but I can see how it would be desirable for beds / chairs too.

One potential complication of adding an 'item required' field to the properties of such items is that it could become confusing to an average player if the interactivity requirements for seemingly identical items are inconsistent from one instance to the next.  Players typically rely on  identically-named items in a game having identical behaviours.  And I would hate for a giant red text scroll to appear at the top of the screen every time I clicked on a chair or a bed for which I wasn't holding the requisite 'item required.'  If the devs do add a 'picked required' field to other items in the sub editor, I _really_ hope that they will get rid of the giant red text scrolls at the top of the screen -- or at least make them optional in the game's settings menu.

Another solution would be if every item in the sub editor had a 'bots should ignore' checkbox.  This would be especially helpful for things like a diving suit locker that is intended for human-only players, or for override switches for doors -- which the bots' AI often mistake for the door controls.  That way we wouldn't have to set an 'ignore target' order in-game for items that are meant exclusively for use by human players.  After all, there is already a 'Non Interactable' flag and a 'Non-Player Team Interactable' checkbox for every item.  Why not have a checkbox that makes items non-interactable only for bots?
